satpy/utils.py

```diff
@@ -841,3 +842,17 @@ def find_in_ancillary(data, dataset):
             f"variables for dataset {data.attrs.get('name')!r}, "
             f"found {cnt:d}")
     return matches[0]
+
+
+def datetime64_to_pydatetime(dt64):
+    """Convert numpy.datetime64 timestamp to Python datetime.
+
+    Discards nanosecond precision, because Python datetime only has microsecond
+    precision.
+
+    Args:
+        dt64 (np.datetime64): Timestamp to be converted
+    Returns (dt.datetime):
+        Converted timestamp
+    """
+    return pd.Timestamp(dt64).to_pydatetime(warn=False)
```

Comment:
What are the cases where this warns that are being ignored? Is there a way to do this without pandas? I know when I've looked this up in the past I think the pandas solution was also the best option, but I'm just curious.

Reply:
Pandas warns when the conversion to datetime loses precision:

```python
In [4]: t = np.datetime64("2000-01-01T00:00:00.000000001")
In [5]: pd.Timestamp(t).to_pydatetime()
<ipython-input-5-1572263aba35>:1: UserWarning: Discarding nonzero nanoseconds in conversion.
  pd.Timestamp(t).to_pydatetime()
Out[5]: datetime.datetime(2000, 1, 1, 0, 0)
```

An alternative solution could be to explicitly cast to datetime precision (microseconds) and then convert to datetime:

```python
In [6]: t.astype("datetime64[us]").astype(dt.datetime)
Out[6]: datetime.datetime(2000, 1, 1, 0, 0)
```

Actually I think that's better, because it's more transparent